lib/manageiq/appliance_console/message_server_configuration.rb

```diff
@@ -0,0 +1,247 @@
+    end
+
+    def ask_for_parameters
+      puts("JJV 0.0 #{File.basename(__FILE__)} / #{__method__}")
+
+      say("\nMessage Server Parameters:\n\n")
+      @username  = ask_for_string("Message Server Username", @username)
+      @password  = ask_for_password("Message Server Password", @password)
+    end
+
+    def show_parameters
+      puts("JJV 0.0 #{File.basename(__FILE__)} / #{__method__}")
+      say("\nMessage Server Configuration:\n")
+      say("Message Server Details:\n")
+      say("  Hostname:      #{@hostname}\n")
+      say("  Username:      #{@username}\n")
+    end
+
+    private
+
+    def get_hostname
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      LinuxAdmin::Hosts.new.hostname
+    end
+
+    def configure_log3
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      FileUtils.cp(@log4_properties_sample_path, @log4_properties_path) unless file_found?(@log4_properties_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_jaas
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        KafkaServer {
+          org.apache.kafka.common.security.plain.PlainLoginModule required
+          username="#{@username}"
+          password="#{@password}"
+          user_admin="#{@password}"
+        };
+      EOS
+
+      File.write(@jaas_config_path, content) unless file_found?(@jaas_config_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_client_properties
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        ssl.truststore.location="#{@truststore_path}"
+        ssl.truststore.password="#{@password}"
+
+        sasl.mechanism=PLAIN
+        security.protocol=SASL_SSL
+        sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \\
+          username="#{@username}" \\
+          password="#{@password}";
+      EOS
+
+      File.write(@client_properties_path, content) unless file_found?(@client_properties_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_ssl
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      content = <<~EOS
+        [ subject_alt_name ]
+      EOS
+
+      File.write(@openssl_config_path, content) unless file_found?(@openssl_config_path)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_logs_directory
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      return true if file_found?(LOGS_DIR)
+
+      FileUtils.mkdir_p(LOGS_DIR)
+      FileUtils.chmod(0744, LOGS_DIR)
+      FileUtils.chown("kafka", "kafka", LOGS_DIR)
+
+      require 'pry'; binding.pry # JJV
+    end
+
+    def configure_firewall
+      say("#{__method__}".gsub("_", " ").titleize)
+
+      AwesomeSpawn.run!("firewall-cmd --add-port=9092/tcp --permanent") # insecure
+      AwesomeSpawn.run!("firewall-cmd --add-port=9093/tcp --permanent") # secure
+      AwesomeSpawn.run!("firewall-cmd --reload")
```

Comment:
Do we do the same thing for postgres?  that is, configure the firewall on the fly?  if so, then :+1:

Reply:
@Fryguy It doesn't appear that we do firewwall config on the fly for postgres. @bdunne had suggested this approach for kafak. @bdunne Can you please comment?